kornia/utils/image.py

```diff
@@ -24,24 +26,50 @@ def image_to_tensor(image: np.ndarray, keepdim: bool = True) -> torch.Tensor:
         raise ValueError(
             "Input size must be a two, three or four dimensional array")
 
-    input_shape = image.shape
     tensor: torch.Tensor = torch.from_numpy(image)
+    # Normalize to channel-last format
+    if len(tensor.shape) == 2:
+        # (H, W) -> (H, W, 1)
+        tensor = tensor.unsqueeze(-1)
+    tensor_shape = tensor.shape
+    tensor = to_bchw(tensor, -1)
+    if keepdim:
+        if len(tensor_shape) == 3:
+            tensor = tensor.reshape(-1, *tensor_shape[:-1])
+        elif len(tensor_shape) == 4:
+            pass
+        else:
+            raise ValueError(f"Unsupported tensor shapes. Got {tensor_shape}.")
+    return tensor
 
-    if len(input_shape) == 2:
-        # (H, W) -> (1, H, W)
-        tensor = tensor.unsqueeze(0)
-    elif len(input_shape) == 3:
-        # (H, W, C) -> (C, H, W)
-        tensor = tensor.permute(2, 0, 1)
-    elif len(input_shape) == 4:
-        # (B, H, W, C) -> (B, C, H, W)
-        tensor = tensor.permute(0, 3, 1, 2)
-        keepdim = True  # no need to unsqueeze
-    else:
-        raise ValueError(
-            "Cannot process image with shape {}".format(input_shape))
```

Comment:
don't remove this piece of code. Seems pretty clear what is doing. What happens later it's not